examples/ui-demo/src/hooks/useRecurringTransactions.ts

```diff
@@ -119,46 +169,36 @@ export const useRecurringTransactions = ({
     const txnHash = await sessionKeyClient
       .waitForUserOperationTransaction(uoHash)
       .catch((e) => {
-        console.log(e);
+        console.error(e);
       });
 
     if (!txnHash) {
-      setCardStatus("initial");
-      return;
+      return handleError(new Error("missing swap txn hash"));
     }
 
-    setTransactions((prev) => {
-      const newState = [...prev];
-      newState[transactionIndex].state = "complete";
-      newState[transactionIndex].externalLink = odyssey.blockExplorers
-        ? `${odyssey.blockExplorers?.default.url}/tx/${txnHash}`
-        : undefined;
-      return newState;
-    });
+    setTransactions((prev) =>
+      prev.map((txn, idx) =>
+        idx === transactionIndex
+          ? {
+              ...txn,
+              state: "complete",
+              externalLink: odyssey.blockExplorers
+                ? `${odyssey.blockExplorers?.default.url}/tx/${txnHash}`
+                : undefined,
+            }
+          : txn
+      )
+    );
   };
 
-  // Mock method to fire transactions for 7702
+  // Mock method to fire transactions
   const handleTransactions = async () => {
     if (!client) {
       console.error("no client");
       return;
     }
 
-    // initial state as referenced by `const initialTransactions` is mutated, so we need to re-create it.
-    setTransactions([
-      {
-        state: "initiating",
-        buyAmountUsdc: 4000,
-      },
-      {
-        state: "initial",
-        buyAmountUsdc: 3500,
-      },
-      {
-        state: "initial",
-        buyAmountUsdc: 4200,
-      },
-    ]);
+    setTransactions(initialTransactions);
```

Comment:
flagging that the reason we did this earlier was that `handleTransaction` mutated the JS object state, so setting it back to the const var `initialState` preserved the mutations that we didn't want to keep.

It looks like this is addressed in this pr by using `.map` in `handleTransaction`, but just to double-check, did you try using the session key more than once with the restart button? The last item should show as "pending", without any of the previous state.

Reply:
actually nvm, I was able to test this out using the vercel preview and it looks good!

Reply:
I like to avoid mutating JS objects in React whenever possible. Makes things a lot more predictable.